sympy/geometry/tests/test_ellipse.py

```diff
@@ -9,6 +9,23 @@
 from sympy import integrate
 from sympy.functions.special.elliptic_integrals import elliptic_e
 
+def test_object_from_equation():
+    from sympy.abc import x, y, a, b
+    assert Circle.object_from_equation(x ** 2 + y ** 2 + 3 * x + 4 * y - 8) == Circle(Point2D(S(-3) / 2, -2),
+                                                                                      sqrt(57) / 2)
+    assert Circle.object_from_equation(x ** 2 + y ** 2 + 6 * x + 8 * y + 25) == Circle(Point2D(-3, -4), 0)
```

Comment:
Do you think this could be a geometric entity function `GeometricEntity.from_equation(eq, x, y)` that would emit the known and matching objects: Circle, Ellipse, Line, etc...

Reply:
@smichr  I think that should be another function which can decide the curve from the equation given. Also, an added functionality for such a function could be giving out an object of that curve (for which it could use the `.object_from_equation` method). But this PR wasn't intended for that. It's a great suggestion, but I would like to take that up as another PR (probably after this one). What do you think?

Reply:
@smichr  What do you think sir?

Reply:
@smichr  Okay I will change the name to `from_equation`, and try to make the code work.
>BTW, are ellipses detected?

Not yet. But I could look into it, and try to implement that along with `line` and `circle`.
Should I do that in this PR itself?

Reply:
> try to implement that along with line and circle

no...let's see if we can get something working for these two, first.

Reply:
@smichr  Sir, I've done that now.
It's looking much better now and for your comment :-
>Line(eq) or Line(eq, x=x, y=y) would convert the equation to the object.

I've added the implementation in such a way that, from now on, if line or ellipse object is needed, the inputs have to be in such a way :-
```
avi@avi-Aspire-A515-51G:~/sympy$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Line
>>> from sympy.abc import *
>>> Line(equation=3 * x + y + 18)
Line2D(Point2D(0, 0), Point2D(1, -3))
>>> Line(equation=3 * a + b + 18, x='a', y='b')
Line2D(Point2D(0, 0), Point2D(1, -3))
```
```
avi@avi-Aspire-A515-51G:~/sympy$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Circle
>>> from sympy.abc import *
>>> Circle(equation = x ** 2 + y ** 2 + 3 * x + 4 * y - 8, x = 'x', y = 'y')
Circle(Point2D(-3/2, -2), sqrt(57)/2)
```
I hope this is what you wanted right?